Gordon360/Services/HousingService.cs

```diff
@@ -68,6 +75,22 @@ public string[] GetAllApartmentHalls()
             return hallsResult.ToArray();
         }
 
+        /// <summary>
+        /// Gets all names of apartment halls
+        /// </summary>
+        /// <returns> AN array of hall names </returns>
+        public string[] GetAllTraditionalHalls()
+        {
+
+            var hallsResult = _context.Housing_Halls.Where(h => h.Type == "Traditional").Select(h => h.Name);
+            if (hallsResult == null || !hallsResult.Any())
+            {
+                throw new ResourceNotFoundException() { ExceptionMessage = "The traditional halls could not be found." };
```

Comment:
```suggestion
                throw new ResourceNotFoundException() { ExceptionMessage = "No traditional halls could be found." };
```
By using "The", you imply that there were traditional halls to be found, but they couldn't be for some reason. Technically this is not the case as there are none in the DB, so it's a little bit misleading.